crates/sozo/ops/src/statistics.rs

```diff
@@ -0,0 +1,187 @@
+use std::fs::{self, File};
+use std::io;
+use std::path::PathBuf;
+
+use anyhow::Result;
+use camino::Utf8PathBuf;
+use starknet::core::types::contract::SierraClass;
+use starknet::core::types::FlattenedSierraClass;
+
+#[derive(Debug, PartialEq)]
+pub struct ContractStatistics {
+    pub contract_name: String,
+    pub number_felts: u64,
+    pub file_size: u64,
+}
+
+fn read_sierra_json_program(file: &File) -> Result<FlattenedSierraClass> {
+    let contract_artifact: SierraClass = serde_json::from_reader(file)?;
+    let contract_artifact: FlattenedSierraClass = contract_artifact.flatten()?;
+
+    Ok(contract_artifact)
+}
+
+fn get_sierra_byte_code_size(contract_artifact: FlattenedSierraClass) -> u64 {
+    contract_artifact.sierra_program.len() as u64
+}
+
+fn get_file_size(file: &File) -> Result<u64, io::Error> {
+    file.metadata().map(|metadata| metadata.len())
+}
+
+fn get_contract_statistics_for_file(
+    file_name: String,
+    sierra_json_file: File,
+    contract_artifact: FlattenedSierraClass,
+) -> ContractStatistics {
+    ContractStatistics {
+        contract_name: file_name,
+        number_felts: get_sierra_byte_code_size(contract_artifact),
+        file_size: get_file_size(&sierra_json_file)
+            .expect(format!("Error getting file size for file").as_str()),
+    }
+}
+
+pub fn get_contract_statistics_for_dir(
+    target_directory: &Utf8PathBuf,
+) -> Result<Vec<ContractStatistics>> {
+    let mut contract_statistics = Vec::new();
+    let target_directory = target_directory.as_str();
+    let built_contract_paths: fs::ReadDir = fs::read_dir(target_directory)
+        .expect(format!("Error reading dir {target_directory}").as_str());
+    for sierra_json_path in built_contract_paths {
+        let sierra_json_path_buff: PathBuf =
+            sierra_json_path.expect("Error getting buffer for file").path();
```

Comment:
```suggestion
        let path: PathBuf = entry?.path()
        
        if path.is_dir() {
            continue;
        }
```

`fs::read_dir` returns the entries in a directory, so need to skip if `path` is a dir